libs/lib-client/diagram/src/lib/diagram/diagram.component.ts

```diff
@@ -62,12 +68,21 @@ export class DiagramComponent implements OnChanges, OnDestroy {
   private updateMatrix() {
     const tileMatrix = makeRenderableMatrix(this.flow, 10, this.isReadOnly);
     this.rowIndexService.updateRowIndexes(tileMatrix);
-    if (tileMatrix.length > 0) {
+    if (tileMatrix.length > 1) {
+      const rowParentsList = this.updateRowParentsList(tileMatrix);
       // matrix is reversed to make sure html stack order always goes from bottom to top
       // i.e. top rows are rendered in front of bottom rows, this ensures branches don't display on top of the tiles above
       this.tileMatrix = tileMatrix.reverse();
-    } else if (!this.isReadOnly) {
+      this.rowParentsList = rowParentsList.reverse();
+    } else if (tileMatrix.length === 0 && !this.isReadOnly) {
       this.tileMatrix = EMPTY_MATRIX;
+    } else {
+      this.tileMatrix = tileMatrix;
     }
   }
+
+  private updateRowParentsList(tileMatrix) {
+    const rowIndexes = this.rowIndexService.getRowIndexes();
```

Comment:
Not sure if exposing the whole rowIndexes to the dragService is a right approach. May be using a higher order function which returns the rowIndex of an asked tile should be sufficient.
```suggestion
    const getRowIndexOf = (taskId: string) => this.rowIndexService.getRowIndexForTask(taskId);
```

Reply:
implemented the suggested change